host/linux/syscall.c

```diff
@@ -1438,3 +1438,26 @@ int oe_syscall_nanosleep_ocall(struct oe_timespec* req, struct oe_timespec* rem)
 
     return nanosleep((struct timespec*)req, (struct timespec*)rem);
 }
+
+/*
+**==============================================================================
+**
+** clock_gettime():
+**
+**==============================================================================
+*/
+
+int oe_syscall_clock_gettime_ocall(int clock_id, struct oe_timespec* cur_time)
+{
+    if (!cur_time)
+        return -1;
+    memset(cur_time, 0, sizeof(struct oe_timespec));
+    if (clock_id != CLOCK_REALTIME)
```

Comment:
Given that we're in the host now, we don't really need this check (and the argument checks above) anymore. Instead, we can just set the `errno` to `0`, call the `clock_gettime`, and return the result directly like other ocall does.

Reply:
Where do we make this checks? I am guessing there should be atleast one place where we need to make this checks.

Reply:
The `CLOCK_REALTIME` argument was hard-coded in the previous `_time()` function (host/linux/time.c). That's why there was a check on the enclave side. I don't think we need the check now with this PR.

Reply:
Also should not we check if cur_time is Null or not.

Reply:
The check is handled by the `clock_gettime` already, therefore we don't need to do additional check.